### PR TITLE
feat(faceted-search): Handle category in badge definitions

### PR DESCRIPTION
**Pull request by vbocquet**
**What is the problem this PR is trying to solve?**
We want to be able to categorize using a new metadata per badge: `category`

**What is the chosen solution to this problem?**
This PR add a new "concept" in the `<AddFacetedPopover />` component: Rows are group by screen:
- A first screen contains with uncategorized badges and category link.
- A screen for each category, with badges of it inside.

Each screen are in the DOM on render, we just use CSS to display or hide some screens. We do that to respect UX animation guide on it.

Because we want to adapt the height of the component based on element on it, we needs some computing about screens height in JS and not in CSS.

Unit tests are update, some new tests cover this feature

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 1]
```diff
@@ -39,53 +66,180 @@ AddFacetRow.propTypes = {
...
+	resetFilter: PropTypes.func.isRequired,
+	onFilter: PropTypes.func.isRequired,
+	filterValue: PropTypes.string.isRequired,
+	t: PropTypes.func.isRequired,
+};
+
+const filterByLabel = label => row => {
+	const rowLabel = isString(row) ? row : row.properties.label;
+
+	return rowLabel.toLowerCase().includes(label);
```
Comment (romainseb) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]:
```suggestion
	return rowLabel.toLowerCase().includes(label.toLowerCase());
```

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 2]
```diff
@@ -39,53 +66,180 @@ AddFacetRow.propTypes = {
...
+};
 
 const AddFacetPopover = ({ badgesDefinitions = [], id, initialFilterValue, onClick, t }) => {
-	const [filterValue, setFilterValue] = useState(initialFilterValue || '');
+	const addFacetId = `${id}-add-facet-popover`;
+
+	const [category, setCategory] = useState(null);
+	const [filterValue, setFilterValue] = useState(initialFilterValue || '')
+
+	const categories = useMemo(
```
Comment (romainseb) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: See how we could change to simplify the render (with some functions extract)

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 3]
```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
...
-							badgeDefinition={badgeDefinition}
-							id={addFacetId}
-							key={badgeDefinition.properties.label}
-							label={badgeDefinition.properties.label}
-							onClick={onClick}
+			<div
+				id={addFacetId}
+				className={theme('tc-add-facet-popover-container')}
+				style={{
+					height: screensHeight[categoryScreenIndex],
```
Comment (jare-talend) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: why height & maxHeight with the same value?

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 4]
```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
...
-	const addFacetId = `${id}-add-facet-popover`;
+
+	const screens = [
+		{
+			category: null,
+			rows: useMemo(
+				() =>
+					[...categories, ...badgesWithoutCategory]
+						.sort(sortByLabel)
+						.filter(filterByLabel(filterValue.toLowerCase().trim())),
```
Comment (jare-talend) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: like @romainseb you can do the  filterValue.toLowerCase() in filterByLabel function

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 5]
```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
...
+				[badgesDefinitions, filterValue]
+			),
+		},
+		...categories.map(categoryName => ({
+			category: categoryName,
+			rows: useMemo(
+				() => badgesDefinitions
+					.filter(badgeDefinition => badgeDefinition.metadata.category === categoryName)
+					.filter(filterByLabel(filterValue.toLowerCase().trim()))
```
Comment (jare-talend) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: too

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 6]
```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
...
+				id={addFacetId}
+				className={theme('tc-add-facet-popover-container')}
+				style={{
+					height: screensHeight[categoryScreenIndex],
+					maxHeight: screensHeight[categoryScreenIndex],
+				}}
+			>
+				{screens.map((screen, index) => (
+					<div
+						ref={screensRef.current[index]}
```
Comment (jare-talend) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: missing key prop

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js [hunk 7]
```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
...
+		() =>
+			badgesDefinitions
+				.filter(badgeDefinition => !!badgeDefinition.metadata.category)
+				.map(badgeDefinition => badgeDefinition.metadata.category)
+				.filter((categoryName, index, arr) => arr.indexOf(categoryName) === index),
+		[badgesDefinitions],
+	);
+
+	const screensCount = categories.length + 1;
+	const [screensHeight, setScreensHeight] = useState(times(screensCount, () => 0));
```
Comment (jare-talend) [inline packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: we can use https://lodash.com/docs/4.17.15#constant
```suggestion
        import constant from 'lodash/constant';
	const [screensHeight, setScreensHeight] = useState(times(screensCount, constant(0)));
```

File: packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js (hunk 3 above)
Comment (vbocquet) [reply packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js]: You're right, fix that

**Comment by vbocquet** [comment]
@romainseb @jare-talend 
Thx pour you're review, i push the changes:
- Tab navigation between screens is fixed (hidden screen are unfocusable now)
- Made a full refactor to simply the render with the using of memoize function
- Fix lowerCase filtering

**Pull request merged**